ray_provider/operators/ray.py

```diff
@@ -274,41 +162,118 @@ def __init__(
+        :return: The Ray dashboard URL if available, None otherwise.
+        """
+        if self.xcom_task_key:
+            parts = self.xcom_task_key.split(".", 1)
+            task: str | None
+            key: str
+            if len(parts) == 2:
+                task, key = parts
+            else:
+                task, key = None, self.xcom_task_key
+
+            ti = context["ti"]
+
+            if task is None:
+                current_task = context["task"]
+                dashboard_url = ti.xcom_pull(task_ids=current_task.task_id, key=key)
+            else:
+                dashboard_url = ti.xcom_pull(task_ids=task, key=key)
+
+            self.log.info(f"Dashboard URL retrieved from XCom: {dashboard_url}")
+            return str(dashboard_url) if dashboard_url is not None else None
+
+        return None
+
+    def _setup_cluster(self, context: Context) -> None:
+        """
+        Set up the Ray cluster if a cluster YAML is provided.
+
+        :param context: The context in which the task is being executed.
+        :raises Exception: If there's an error during cluster setup.
+        """
+        try:
+            if self.ray_cluster_yaml:
+                self.hook.setup_ray_cluster(
+                    context=context,
+                    ray_cluster_yaml=self.ray_cluster_yaml,
+                    kuberay_version=self.kuberay_version,
+                    gpu_device_plugin_yaml=self.gpu_device_plugin_yaml,
+                    update_if_exists=self.update_if_exists,
+                )
+        except Exception as e:
+            raise e
+
+    def _delete_cluster(self) -> None:
+        """
+        Delete the Ray cluster if a cluster YAML is provided.
+
+        :raises Exception: If there's an error during cluster deletion.
+        """
+        try:
+            if self.ray_cluster_yaml:
+                self.hook.delete_ray_cluster(
+                    ray_cluster_yaml=self.ray_cluster_yaml,
+                    gpu_device_plugin_yaml=self.gpu_device_plugin_yaml,
+                )
+        except Exception as e:
+            raise e
 
     def execute(self, context: Context) -> str:
         """
         Execute the Ray job submission and monitoring.
 
+        This method submits the Ray job to the cluster and, if configured to wait for completion,
+        monitors the job status until it reaches a terminal state or times out.
+
         :param context: The context in which the task is being executed.
         :return: The job ID of the submitted Ray job.
         :raises AirflowException: If the job fails, is cancelled, or reaches an unexpected state.
         """
-        if self.xcom_task_key:
-            task, key = self.xcom_task_key.split(".")
-            ti = context["ti"]
-            self.dashboard_url = ti.xcom_pull(task_ids=task, key=key)
-            self.log.info(f"Dashboard URL retrieved from XCom: {self.dashboard_url}")
+
+        self._setup_cluster(context=context)
```

Comment:
We should give an option to run job on existing cluster

Reply:
That option already exists..

See Example 1 in the code samples doc (from this PR). User will just not pass in the ray_cluster_yaml input parameter